src/main/java/teammates/ui/webapi/action/BasicFeedbackSubmissionAction.java

```diff
@@ -99,6 +102,9 @@ protected void checkAccessControlForStudentFeedbackSubmission(
      * Gets the instructor involved in the submission process.
      */
     protected InstructorAttributes getInstructorOfCourseFromRequest(String courseId) {
+        if (userInfo == null || !userInfo.isInstructor) {
```

Comment:
Same reason here. 

> Actually, I think it is not the method's responsibility to verify the instructor privilege here. As long as the pre-condition for `getInstructorOfCourseFromRequest` is fulfilled, it should executed.

- The login check should be only verified at L110 (where you delete the old code)
- The caller method should verify that the entity is instructor (I think already is as `checkAccessControlForInstructorFeedbackSubmission`).